app/uk/gov/hmrc/transitmovementsrouter/connectors/EISConnector.scala

```diff
@@ -89,10 +92,16 @@ class EISConnectorImpl(
       (t: Either[RoutingError, Unit]) => Future.successful(t.isRight),
       onFailure
     ) {
+      val data          = messageSender.value.split("-")
+      var correlationId = UUID.randomUUID().toString
+      if (data.size == 2) {
+        correlationId = CorrelationId(MovementId(data(0)), MessageId(data(1))).value
+      }
       val requestHeaders = hc.headers(OutgoingHeaders.headers) ++ Seq(
-        "X-Correlation-Id"        -> UUID.randomUUID().toString,
+        "X-Correlation-Id"        -> correlationId,
         "CustomProcessHost"       -> "Digital",
         "X-Message-Sender"        -> messageSender.value,
```

Comment:
"X-Message-Sender" is repeated, we set it here and also is present in OutgoingHeaders.

Reply:
I am deleting. from OutgoingHeaders as we construct messageSender in the router app